PolySTAR-Taproot-project/src/communication/cv_serial_data.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef CV_SERIAL_DATA_HPP_
+#define CV_SERIAL_DATA_HPP_
+
+#include "tap/communication/serial/ref_serial_data.hpp"
+#include "tap/util_macros.hpp"
+
+using tap::communication::serial::RefSerialData;
+using tap::communication::serial::Uart;
+
+namespace src::communication::cv
+{
+/**
+ * Contains enum and struct definitions used in the `CVSerial` and `CVHandler` classes
+ * 
+ */
+class CVSerialData
+{
+public:
+    class Tx
+    {
+    public:
+        // Outgoing message IDs
+        enum MESSAGE_ID : uint8_t {
+            GAME_STATUS_MESSAGE = 0x01,
+            EVENT_MESSAGE = 0x02,
+            TURRET_FEEDBACK_MESSAGE = 0x03,
+            POSITION_FEEDBACK_MESSAGE = 0x04
+        };
+
+        // Message lengths in bytes
+        enum DATA_LEN : uint8_t {

```

Comment:
Consider coupling the size of a message with the size of the struct itself. The message should know its size (`constepxr` static member + `static_assert` in constructor ?).
Also an enum does not convey the meaning of a size (what if two messages happen to be the same length ?)